src/file.ts

```diff
@@ -3568,11 +3561,15 @@ class File extends ServiceObject<File> {
       options
     );
 
+    const apiEndpoint = this.storage.apiEndpoint;
+    const bucketName = this.bucket.name;
+    const uri = `https://${apiEndpoint}/upload/storage/v1/b/${bucketName}/o`;
```

Comment:
```suggestion
    const uri = `${this.storage.url}/upload/storage/v1/b/${bucketName}/o`;
```

This ensures host protocols other than `https://` work here, in conjunction with googleapis/gcs-resumable-upload#349.